### PR TITLE
Initial working role 

### PR DESCRIPTION
**Pull request by fujexo**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-  :target: https://travis-ci.org/adfinis-sygroup/ansible-role-_template
+.. image:: https://img.shields.io/travis/adfinis-sygroup/ansible-role-icinga2_master.svg?style=flat-square
+  :target: https://travis-ci.org/adfinis-sygroup/ansible-role-icinga2_master
 
-.. image:: https://img.shields.io/badge/galaxy-adfinis--sygroup._template-660198.svg?style=flat-square
-  :target: https://galaxy.ansible.com/adfinis-sygroup/_template
+.. image:: https://img.shields.io/badge/galaxy-adfinis--sygroup.icinga2_master-660198.svg?style=flat-square
+  :target: https://galaxy.ansible.com/adfinis-sygroup/icinga2_master
 
 A brief description of the role goes here.
```
Comment (hairmare) [inline README.rst]: > A brief description of the role goes here.

:sob:

File: tasks/configuration.yml [hunk 2]
```diff
@@ -1 +1,157 @@
...
+
+- name: execute icinga2 api setup command
+  command: icinga2 api setup
+  args:
+    creates: /var/lib/icinga2/ca/ca.crt
+  notify: icinga2_master reload icinga2
+
+- name: configure icinga2 api feature
+  template:
+    src: templates/etc/icinga2/features-enabled/api.conf.j2
```
Comment (MrStrix) [inline tasks/configuration.yml]: put the file in `features-available` and link it to `features-enabled`

File: tasks/configuration.yml [hunk 3]
```diff
@@ -1 +1,157 @@
...
+    seuser: system_u
+    serole: object_r
+    setype: icinga2_etc_t
+    selevel: s0
+    state: directory
+
+- name: create global template configurations
+  template:
+    src: templates/etc/icinga2/zones.d/global-templates/{{ item }}.j2
+    dest: /etc/icinga2/zones.d/global-templates/{{ item}}
```
Comment (MrStrix) [inline tasks/configuration.yml]:
```suggestion
     dest: /etc/icinga2/zones.d/global-templates/{{ item }}
```

**Comment by MrStrix** [comment]
ansible-lint fails because of long lines. as the affected lines are paths, we do not have a reliable way to use multi-line json. how to skip tasks: https://github.com/ansible/ansible-lint/issues/132

**Comment by MrStrix** [comment]
ansible-lint will always fail with this error message: https://github.com/tsukinowasha/ansible-lint-rules/issues/5

File: README.rst [hunk 4]
```diff
@@ -35,34 +36,31 @@ should be mentioned here as well.
 Dependencies
```
Comment (karras) [inline README.rst]: Describe Requirements and Role Variables...

File: handlers/main.yml [hunk 5]
```diff
@@ -1 +1,6 @@
 ---
+
+- name: icinga2_master reload icinga2
+  service:
+    name: icinga2.service
```
Comment (karras) [inline handlers/main.yml]: Shouldn't we use only `icinga2` as the service name thus it's independent from systemd?

File: tasks/configuration.yml [hunk 6]
```diff
@@ -1 +1,163 @@
...
+    selevel: s0
+  notify: icinga2_master reload icinga2
+
+- name: create zone configuration directory for each host
+  file:
+    path: "/etc/icinga2/zones.d/{{ hostvars[item].icinga2_client_parent_zone }}"
+    state: directory
+    owner: icinga
+    group: icinga
+    mode: 0755
```
Comment (karras) [inline tasks/configuration.yml]: Isn't `0644` sufficient for the files?

File: tasks/configuration.yml [hunk 7]
```diff
@@ -1 +1,163 @@
...
+    dest: /etc/icinga2/features-enabled/api.conf
+    state: link
+
+# Global Icinga2 configuration
+- name: ensure the global-templates directory is present
+  file:
+    dest: /etc/icinga2/zones.d/global-templates
+    owner: icinga
+    group: icinga
+    mode: 0755
```
Comment (karras) [inline tasks/configuration.yml]: Isn't `0644` sufficient for the files?

File: tasks/configuration.yml [hunk 8]
```diff
@@ -1 +1,163 @@
...
+    seuser: system_u
+    serole: object_r
+    setype: icinga2_etc_t
+    selevel: s0
+  loop: "{{ groups['monitoring-client'] }}"
+  notify: icinga2_master reload icinga2
+
+- name: ensure that icinga2 is started and enabled on boot
+  service:
+    name: icinga2.service
```
Comment (karras) [inline tasks/configuration.yml]: s/icinga2.service/icinga2/

File: tasks/main.yml [hunk 9]
```diff
@@ -6,16 +6,11 @@
...
-    - 'role::$role'
-    - 'role::$role:install'
-    - 'role::$role:config'
-
-- import_tasks: installation.yml
-  tags:
-    - 'role::$role'
-    - 'role::$role:install'
+    - 'role::icinga2_master'
+    - 'role::icinga2_master:install'
```
Comment (karras) [inline tasks/main.yml]: Remove the install tag

File: templates/etc/icinga2/features-available/api.conf.j2 [hunk 10]
```diff
@@ -0,0 +1,14 @@
+/**
+ *  {{ ansible_managed }}
```
Comment (karras) [inline templates/etc/icinga2/features-available/api.conf.j2]: Version reference?

File: templates/etc/icinga2/zones.d/global-templates/templates.conf.j2 [hunk 11]
```diff
@@ -0,0 +1,83 @@
+/*
+ * Generic template examples.
+ */
```
Comment (karras) [inline templates/etc/icinga2/zones.d/global-templates/templates.conf.j2]: Version reference? Ansible managed?

File: templates/etc/icinga2/zones.d/global-templates/services.conf.j2 [hunk 12]
```diff
@@ -0,0 +1,114 @@
+/*
+ * Service apply rules.
```
Comment (karras) [inline templates/etc/icinga2/zones.d/global-templates/services.conf.j2]: Version reference? Ansible managed?

File: templates/etc/icinga2/zones.d/global-templates/commands.conf.j2 [hunk 13]
```diff
@@ -0,0 +1,186 @@
+/* Command objects */
+
```
Comment (karras) [inline templates/etc/icinga2/zones.d/global-templates/commands.conf.j2]: Version reference? Ansible managed?

File: tasks/configuration.yml (hunk 6 above)
Comment (fujexo) [reply tasks/configuration.yml]: This state creates a directory :)

File: tasks/configuration.yml (hunk 7 above)
Comment (fujexo) [reply tasks/configuration.yml]: This task creates a directory :)

**Pull request merged**